src/serac/physics/solid.hpp

```diff
@@ -326,6 +338,19 @@ class Solid : public BasePhysics {
    */
   void advanceTimestep(double& dt) override;
 
+  /**
+   * @brief Solve the adjoint problem
+   * @note It is expected that the forward analysis is complete and the current displacement state is valid
+   * @note If the essential boundary state is not specified, homogeneous essential boundary conditions are applied
+   *
+   * @param[in] adjoint_load_form The linear form that when assembled contains the right hand side of the adjoint system
+   * @param[in] state_with_essential_boundary A optional finite element state containing the non-homogenous essential
+   * boundary condition data for the adjoint problem
+   * @return The computed adjoint finite element state
+   */
+  virtual const serac::FiniteElementState& solveAdjoint(mfem::ParLinearForm& adjoint_load_form,
+                                                        FiniteElementState*  state_with_essential_boundary = nullptr);
```

Comment:
I forgot if we've had arrived at a convention whether `state_with_essential_boundary` should be `std::optional` or not? There are definitely some pros and cons.

Reply:
The problem is there is no such thing as an optional reference and I would like to avoid copying if possible. Is there a preferred solution for this common problem @joshessman-llnl ?